### ISSUE TITLE
Forward scroll slower than Backward scroll motion in Infinite Dolly mode

### ISSUE THREAD

**Issue by diemildefreude** [yomotsu/camera-controls#525]
### Describe the bug

I'm on using the trackpad of a Lenovo Legion laptop with Windows 11. If I do one backwards swipe, it takes me about three forward swipes to get back where I started: [infinite dolly example](https://yomotsu.github.io/camera-controls/examples/infinity-dolly.html)

This issue doesn't seem to be present in the basic mode: [basic example](https://yomotsu.github.io/camera-controls/examples/basic.html )

Maybe related to #497 and #499 ?

### Device

Desktop

### OS

Windows

### Browser

Edge

**Comment by aka-blackboots** [comment]
@diemildefreude, were you able to solve this? 🤔

**Comment by diemildefreude** [comment]
> @diemildefreude, were you able to solve this? 🤔

Nope! For a little while I tried changing the dollySpeed based on scroll/pinch direction, but the change would get applied too late so you'd see a stutter on direction change. I've posted about a couple of related issues on here, but the developer doesn't seem to be very active here lately.

Anyway, I've learned to live with it. xD

**Comment by LiteralGenie** [comment]
So changing [these two lines](https://github.com/yomotsu/camera-controls/blob/75d368a4b3dbb09a88582e38af48fb7591fd91a6/src/CameraControls.ts#L3153-L3154) to use `_zoomEnd` instead of `zoom` seems to fix the issue for me.
```ts
protected _zoomInternal = ( delta: number, x: number, y: number ): void => {
	const zoomScale = Math.pow( 0.95, delta * this.dollySpeed );
	// const lastZoom = this._zoom;
	// const zoom = this._zoom * zoomScale;
	const lastZoom = this._zoomEnd;
	const zoom = this._zoomEnd * zoomScale;
}
```

tldr on the why is that scroll events firing too quickly can interrupt each other and make the final zoom value unpredictable.

This can be tested using the below snippet, where zooming in (`e`) then out (`q`) should start and end on the same zoom value. Without the above changes, the bug happens when `eventDelay < CameraControls.draggingSmoothTime`.
```ts
const deltaSize = 100
const eventDelay = 50
const eventCount = 10

const dispatch = (deltaY: number) => {
    controls._domElement.dispatchEvent(
        new WheelEvent('wheel', {
            clientX: -1,
            clientY: -1,
            deltaMode: 0,
            deltaX: 0,
            deltaY,
        }),
    )
}

window.addEventListener('keydown', async (ev) => {
    let sign: number
    if (ev.key === 'q') {
        sign = -1
    } else if (ev.key === 'e') {
        sign = 1
    } else {
        return
    }

    console.log(ev.key === 'q' ? 'zooming IN' : 'zooming OUT')
    console.log('start', controls._zoom.toFixed(4))

    for (let idx = 0; idx < eventCount; idx++) {
        dispatch(sign * deltaSize)
        await sleep(eventDelay)
    }

    await sleep(500)
    console.log('end', controls._zoom.toFixed(4))

    async function sleep(t: number) {
        return new Promise((r) => setTimeout(r, t))
    }
})
```

---

Longer explanation:

From what I understand the controls use both a `this._zoom` and `this._zoomEnd` value. The former represents the current zoom value and the latter represents the desired zoom value. Whenever `CameraControls.update(clock.getDelta())` is called, `this._zoom` gets a bit closer to `this._zoomEnd` via a smoothing [function](https://github.com/yomotsu/camera-controls/blob/75d368a4b3dbb09a88582e38af48fb7591fd91a6/src/CameraControls.ts#L2633-L2634).

The `_zoomInternal()` function that's modified above is the [event handler](https://github.com/yomotsu/camera-controls/blob/75d368a4b3dbb09a88582e38af48fb7591fd91a6/src/CameraControls.ts#L805-L811) for scroll events and its role is [updating](https://github.com/yomotsu/camera-controls/blob/75d368a4b3dbb09a88582e38af48fb7591fd91a6/src/CameraControls.ts#L1753) `this._zoomEnd` based on the distance scrolled (`delta`, which is [WheelEvent.deltaY](https://developer.mozilla.org/en-US/docs/Web/API/WheelEvent)). However, because it was using `this._zoom` instead of `this._zoomEnd`, the neext `_zoomEnd` depends on how far the smoothing has progressed, aka the time since last scroll event. Or in other words, scroll events can interrupt each other.

Also of note is that this only happens with an orthographic camera because with a perspective camera, the mouse wheel defaults to the dolly functionality instead of zoom. And the [dolly function](https://github.com/yomotsu/camera-controls/blob/75d368a4b3dbb09a88582e38af48fb7591fd91a6/src/CameraControls.ts#L3118-L3120) behaves like above, using its `_zoomEnd` equivalent instead of the mid-smoothing value.
```ts
protected _dollyInternal = ( delta: number, x: number, y : number ): void => {
	const dollyScale = Math.pow( 0.95, - delta * this.dollySpeed );
	const lastDistance = this._sphericalEnd.radius;
	const distance = this._sphericalEnd.radius * dollyScale;
	const clampedDistance = clamp( distance, this.minDistance, this.maxDistance );
	const overflowedDistance = clampedDistance - distance;
```

And finally, the reason forward scrolling feels different than backward scrolling is probably just human bias. As in scrolling backward (zooming out / positive event delta) seems to [generate](https://github.com/user-attachments/assets/e687b1ee-075b-481e-a6e6-9a0a8c05596b) spikier deltas and I guess this is exaggerated by how the smoothing function is non-linear or something. idk, haven't looked that hard at the math.
